app/models/user.rb

```diff
@@ -7,10 +7,8 @@ class User < ApplicationRecord
 
   def self.search(query)
     if query.present?
-      joins(:suppliers).where(
-        'users.name ILIKE :query OR email ILIKE :query OR suppliers.name ILIKE :query',
-        query: "%#{query}%"
-      )
+      eager_load(:suppliers)
```

Comment:
They don't make these method names logical, do they?

Reply:
yep, it's very confusing. this is a good summary of all the various ins and outs - http://blog.scoutapp.com/articles/2017/01/24/activerecord-includes-vs-joins-vs-preload-vs-eager_load-when-and-where